src/umd/ReactUMDEntry.js

```diff
@@ -17,17 +17,8 @@ var React = require('React');
 var ReactUMDEntry = Object.assign({
   __SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED: {
     ReactCurrentOwner: require('react/lib/ReactCurrentOwner'),
+    ReactComponentTreeHook: require('react/lib/ReactComponentTreeHook'),
```

Comment:
It seems to me that if you just split up the ReactComponentTreeHook module into smaller pieces in the shared folder that both ReactComponentTreeHook and the Fiber prod stack thing can depend on the rest will fall naturally from there. ReactComponentTreeHook gets disabled in prod and so the dependencies fall off. Only the small dependencies that you split out (getStackAddendumByWorkInProgressFiber) and its dependencies get embedded in the renderer production package.